addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2244,6 +2245,13 @@ var SnippetsMenu = Widget.extend({
         exclude += `${exclude && ', '}.o_snippet_not_selectable, .o_not_editable :not([contenteditable="true"])`;
 
         let filterFunc = function () {
+            // Here we have to make sure that we return false if the element
+            // matches the data-exclude from the XML option declaration. But if
+            // it's not the case, we return true if the element is an editable
+            // media even if it matches the exclude selector.
+            if ($(this).is(dataExclude)) {
+                return false;
+            }
```

Comment:
I don't understand your fix @Guillaume-gdi ... why would you need the "original exclude"? Why would he behave differently from the "fixed exclude"? :thinking: 

Is not the fix here just about inverting the condition?

So going from:
```javascript
            if (!$(this).is(exclude)) {
                return true;
            }
            if (this.classList.contains('o_editable_media')) {
                return shouldEditableMediaBeEditable(this);
            }
            return false;
```
to:
```javascript
            if ($(this).is(exclude)) {
                return false;
            }
            if (this.classList.contains('o_editable_media')) {
                return shouldEditableMediaBeEditable(this);
            }
            return true;
```

Reply:
@qsm-odoo Your proposal would break the team block again.. The options on the images would not appear :cry:.
Just before the `filterFunc` there is this line
```js
exclude += `${exclude && ', '}.o_snippet_not_selectable, .o_not_editable :not([contenteditable="true"])`;
```
that prevent the options of elements in a not editable parent from appearing (even with the `o_editable_media`). That's the reason I need the  "original exclude"

Let me know if it is not clear. I can improve my comment :+1:

Reply:
Oh ok, I got half-confused with your comment here https://github.com/odoo/odoo/pull/132849/commits/88ffb89e5f0e37f14dc1390ac008b59544eaca0f#r1303007391 my bad.

In fact the issue really comes from the fact the o_editable_media is meant to bypass o_not_editable but not data-exclude and the rest. In fact, in this stuff: `exclude += `${exclude && ', '}.o_snippet_not_selectable, .o_not_editable :not([contenteditable="true"])`;`, it should bypass the o_not_editable part but not that o_snippet_not_selectable AFAIK. Let's make this clearer:
- Invert the condition like I wanted to do (so "if to exclude, return false, else return true")
- Keep the "o_snippet_not_selectable" part added to the exclude selector outside of the filterFunc function
- Add the o_not_editable part inside the filterFunc, after the editable_media check.
- Comment each step, that way we have something clear:
    1) Exclude what the caller asked to exclude
    2) Also exclude non editable elements, except for media explicitly bypassing that

Reply:
It's done @qsm-odoo :+1: